libs/openFrameworks/gl/ofShader.h

```diff
@@ -127,6 +127,38 @@ class ofShader {
 	
 	void setUniforms(const ofParameterGroup & parameters) const;
 
+    /// Changes a numerical define value
+    ///
+    /// Allows to change a define in the shader declared like
+    ///
+    /// #define SOME_FEATURE 1
+    ///
+    /// calling setDefineConstant("SOME_FEATURE", 0);
+    ///
+    /// Be aware that this recompiles the shader which can be
+    /// slow if done often. This is only recommendable for
+    /// values that don't change very often and if they
+    /// were a uniform would introduce runtime evaluated
+    /// conditionals or similar which would slow down the shader
+    void setDefineConstant(const string & name, float value);
+    void setDefineConstant(const string & name, int value);
+    void setDefineConstant(const string & name, bool value);
+
+
+    /// Changes a numerical define value
+    ///
+    /// Allows to change a define in the shader declared like
+    ///
+    /// const float somevalue=10;
+    ///
+    /// calling setConstant1f("somevalue", 20.f);
+    ///
+    /// Be aware that this recompiles the shader which can be
+    /// slow if done often.
+    void setConstant1f(const string & name, float value);
```

Comment:
It's not just that it recompiles the shader - it also globally invalidates the previous GL shader program object across all users of this shader if I understand correctly. This could be a quite surprising side-effect =)

I wonder if it wasn't possible to make these methods only callable before calling setup() - or to print out a performance warning otherwise - as some folks might be tempted to call these method in the draw() method repeatedly on a shader.

Maybe also changing the name from `setConstant..` to something more like `setSourceCodeConstant` might help to distinguish these methods from the classic "draw" `setUniform..` methods...